vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java

```diff
@@ -837,12 +850,20 @@ private void registerValidation() {
         if (validationRegistration != null) {
             validationRegistration.remove();
         }
-        validationRegistration = getElement().addPropertyChangeListener("value",
+        validationRegistration = getElement().addPropertyChangeListener(VALUE_PROPERTY_NAME,
                 validationListener);
     }
 
     private void runBeforeClientResponse(SerializableConsumer<UI> command) {
         getElement().getNode().runWhenAttached(ui -> ui
                 .beforeClientResponse(this, context -> command.accept(ui)));
     }
+
+    private void fireSizeEvent() {
+        final int newDataSize = getDataProvider().size(new Query<>());
```

Comment:
Do we need the query here as we can just use the previous size collection as an variable that is updated for each reset.
e.g. `AtomicInteger newSize` which would update on reset and then be used here

Reply:
Yes, good catch!